lib/astc_encode.cpp

```diff
@@ -610,7 +611,7 @@ ktxTexture2_CompressAstcEx(ktxTexture2* This, ktxAstcParams* params) {
                         block_size_x, block_size_y, block_size_z);
     quality = astcQuality(params->qualityLevel);
     profile = astcEncoderAction(*params, BDB);
-    swizzle = astcSwizzle(*params);
+    swizzle = (params->normalMap ? swizzle_normal : astcSwizzle(*params));
```

Comment:
Is overriding `params->inputSwizzle` the right thing? Or should we allow inputSwizzle to override swizzle_normal? The latter would provide more flexibility. Whatever we decide, it needs to be documented. Currently it does not say that normalMap overrides inputSwizzle.

We need need to add similar code in `basis_encode.cpp`.

Reply:
I am not sure this works. For the normal mode the encoder expects the swizzle to be `ASTCENC_SWZ_R, ASTCENC_SWZ_R, ASTCENC_SWZ_R, ASTCENC_SWZ_G`. I don't see a way to give it a "normal_swizzle" and another swizzle on top of that so we must override `inputSwizzle` if `normal_mode` requested.

I guess what happens with inputSwizzle before it reaches the encoder is still worth looking into. Would the data reaching the encoder be already swizzled with the `inputSwizzle`? In the astc_encoder case I assume not thats why we pass that onto the encoder. Is it swizzled early before it hits basis_encoder? If yes then we can't generally say it overrides it but in astc case it must.

Reply:
I'm not suggesting trying to use both swizzles. Sorry it wasn't clear. I am asking which should have priority

1. always do the normal swizzle as we are now
2. use inputSwizzle, if set, otherwise do the "normal" swizzle.

2 gives flexibility but opens the possibility of shooting oneself in the foot.

For the BasisU encoder inputSwizzle is done in the encoder as it also is for the ASTC encoder. I am not suggesting changing that. toktx only handles input swizzles for non-compressed formats.

I'm leaning towards 2. It is easy to implement so even if nobody ever uses it, it is not a big waste.

Reply:
I looked into this. It looks like we are setting inputSwizzle before it hits the encode phase to either `defaultSwizzle` or `inputSwizzle` or undefined swizzle. At the encoder stage in `astc_encode.cpp` we don't know if it was set or default set or undefined and convert it to astc swizzle unconditionaly `for (int i = 0; i < 4; i++) {`: Thats why I have moved the logic into `toktx.cc`. If normal mode is set we set the options.swizzle accordingly.